src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/products.html.twig

```diff
@@ -74,60 +76,138 @@
+            {% if not orderForViewing.delivered %}
+              <td class="text-right">
+                <a href="#"
+                   class="tooltip-link"
+                   data-toggle="pstooltip"
+                   data-placement="top"
+                   data-original-title="{{ 'Edit'|trans({}, 'Admin.Actions') }}"
+                >
+                  <i class="material-icons js-order-product-update-btn"
+                     data-update-url="{{ path('admin_orders_update_product', {'orderId': orderForViewing.id, 'orderDetailId': product.orderDetailId}) }}"
+                     data-product-name="{{ product.name }}"
+                     data-product-quantity="{{ product.quantity }}"
+                     data-product-price-tax-incl="{{ product.unitPriceTaxInclRaw }}"
+                     data-product-price-tax-excl="{{ product.unitPriceTaxExclRaw }}"
+                     data-toggle="modal"
+                     data-target="#updateOrderProductModal"
+                  >edit</i>
+                </a>
 
-              <a href="#"
-                 class="ml-2 tooltip-link"
-                 data-toggle="pstooltip"
-                 data-placement="top"
-                 data-original-title="{{ 'Delete'|trans({}, 'Admin.Actions') }}"
-              >
-                <i class="material-icons">delete</i>
-              </a>
-            </td>
+                <a href="#"
+                   class="ml-2 tooltip-link"
+                   data-toggle="pstooltip"
+                   data-placement="top"
+                   data-original-title="{{ 'Delete'|trans({}, 'Admin.Actions') }}"
+                >
+                  <i class="material-icons">delete</i>
+                </a>
+              </td>
+            {% endif %}
           </tr>
         {% endfor %}
       </tbody>
     </table>
 
     <div class="row mb-3">
       <div class="col-md-12 text-right">
-        <button type="button" class="btn btn-outline-secondary">
-          <i class="material-icons">add_circle_outline</i>
-          {{ 'Add a product'|trans({}, 'Admin.Orderscustomers.Feature') }}
-        </button>
+        {% if not orderForViewing.delivered %}
+          <button type="button" class="btn btn-outline-secondary" data-toggle="modal" data-target="#addOrderProductModal">
+            <i class="material-icons">add_circle_outline</i>
+            {{ 'Add a product'|trans({}, 'Admin.Orderscustomers.Feature') }}
+          </button>
+        {% endif %}
         <button type="button" class="btn btn-outline-secondary" data-toggle="modal" data-target="#addOrderDiscountModal">
           <i class="material-icons">confirmation_number</i>
           {{ 'Add a new discount'|trans({}, 'Admin.Orderscustomers.Feature') }}
         </button>
       </div>
     </div>
 
+    {% if orderForViewing.discounts.discounts is not empty %}
+      <table class="table">
+        <thead>
+          <tr>
+            <th>{{ 'Discount name'|trans({}, 'Admin.Global') }}</th>
```

Comment:
I only have _Name_ (_Admin.Global_) in the mockups. cf. https://projects.invisionapp.com/d/main?origin=v7#/console/18147703/381874231/preview

Reply:
That's right, I think I have followed legacy wording and I think it makes sense, otherwise, it requires user to what "name" is for, it's not explicitly marked that these are discounts. Wdyt? :)

Reply:
I agree but it might look weird to talk about cart rule before and discount after, doesn't it?

Reply:
Yup, it might, but this wording is mixed a lot. Even since 1.7 side navigation was renamed from "Cart rules" (as it was in 1.6) to "Discounts" (here https://prnt.sc/pje0gk). 

So how do we deal with it? :thinking:

Reply:
In the back office, 'Discounts' is a generic word that refers to both cart rules and catalog price rules. If the first word is very clear for all users, it is quite uneasy to get at first sight what designate cart rules and catalog price rules. If the discount here only regards the possibility to add a cart rule, I suggest to call it cart rule:

- it is precise enough and familiarizes users to our discount terminology 
- it will also help users know what kind of discount to apply here

So, to summarize, I've just talked to @MatShir and here is what we recommend: using the button 'Add a discount' (in Admin.Orderscustomers.Feature) on the Order Details page, then calling the pop-up frame 'Add new cart rule' (in Admin.Catalog.Feature) and keeping the first field as 'Name' (in Admin.Global).

Reply:
Done.